libs/vkd3d/command.c

```diff
@@ -1508,6 +1509,43 @@ static ULONG STDMETHODCALLTYPE d3d12_shared_fence_AddRef(d3d12_fence_iface *ifac
     return refcount;
 }
 
+static void d3d12_shared_fence_open_kmt(struct d3d12_shared_fence *fence, struct d3d12_device *device)
+{
+#ifdef _WIN32
+    const struct vkd3d_vk_device_procs *vk_procs = &device->vk_procs;
+    VkSemaphoreGetWin32HandleInfoKHR win32_handle_info;
+    D3DKMT_OPENSYNCOBJECTFROMNTHANDLE open = {0};
+    NTSTATUS status;
+    VkResult vr;
+
+    win32_handle_info.sType = VK_STRUCTURE_TYPE_SEMAPHORE_GET_WIN32_HANDLE_INFO_KHR;
+    win32_handle_info.pNext = NULL;
+    win32_handle_info.semaphore = fence->timeline_semaphore;
+    win32_handle_info.handleType = VK_EXTERNAL_SEMAPHORE_HANDLE_TYPE_D3D12_FENCE_BIT;
+
+    if ((vr = VK_CALL(vkGetSemaphoreWin32HandleKHR(device->vk_device, &win32_handle_info, &open.hNtHandle))))
+        ERR("Failed to get exported semaphore handle, vr %d.\n", vr);
+    else
+    {
+        if ((status = D3DKMTOpenSyncObjectFromNtHandle(&open)))
```

Comment:
If we're holding onto a KMT object here, what does that accomplish for a sync object which has no external metadata? I thought the HANDLE itself would be enough.

Reply:
See elsewhere, local handles are required for the D3DKMTShareObjects calls, to create the actual Win32 shared handles (for the object name, rights, attributes etc...).

Reply:
Ok, so in theory they could be created on-demand every time there's a CreateSharedHandle() call, and destroyed after exporting? It's fine to keep the object around, but probably good to comment why the object lingers for non-obvious reasons.

Reply:
Found a bug in this area for legacy path.